### PR TITLE
Ported recall_bscore

### PR DESCRIPTION
**Pull request by Eman22S**

### CODE REVIEW

**Comment by kasimebrahim** [comment]
Hey @Eman22S till @ngeiswei is back and give you the real comments, here are a few things,
1 in discriminator::count you might want to remove the unnecessary white-spaces and the new line after if (...).
2 you might also shorten the code using transform.
3 the indentation is not consistent in the test
4 the test author doesn't say eman

**Comment by Eman22S** [comment]
> Hey @Eman22S till @ngeiswei is back and give you the real comments, here are a few things,
> 1 in discriminator::count you might want to remove the unnecessary white-spaces and the new line after if (...).
ok fixed that!!
> 2 you might also shorten the code using transform.
> 3 the indentation is not consistent in the test
> 4 the test author doesn't say eman
This one I have fixed it already check it now

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
-
+INCLUDE_DIRECTORIES(
+        ${CMAKE_CURRENT_SOURCE_DIR}
+        ${CMAKE_CURRENT_BINARY_DIR}
+        ${CMAKE_BINARY_DIR}
+)
```
Comment (ngeiswei) [inline examples/CMakeLists.txt]: You can rebase your branch onto singnet/master as this is already in there.

File: opencog/moses/scoring/discriminating_bscore.cc [hunk 2]
```diff
@@ -195,6 +246,43 @@ vector<discriminator::d_counts> discriminator::counts(const combo_tree& tr) cons
...
+    for (const CTable::value_type& vct : _ctable) {
+        // vct.first = input vector
+        // vct.second = counter of outputs
+
+        double pos = sum_true(vct.second);
+        double neg = sum_false(vct.second);
+        unsigned totalc = vct.second.total_count();
+
+        d_counts ctr;
+        if (HandleCast(link_result.at(i))->get_type() == TRUE_LINK)
```
Comment (ngeiswei) [inline opencog/moses/scoring/discriminating_bscore.cc]: Better use `bool_value_to_bool` define in

https://github.com/singnet/asmoses/blob/998557895ad4e39ea43a80fbef2f5eb75351640c/opencog/utils/valueUtils.h#L12-L15

that way if we change the representation of Boolean values that code won't change.

File: opencog/moses/scoring/discriminating_bscore.cc [hunk 3]
```diff
@@ -490,6 +604,9 @@ prerec_bscore::prerec_bscore(const CTable& ct,
 
 // Nearly identical to recall_bscore, except that the roles of precision
 // and recall are switched.
+
+
+
```
Comment (ngeiswei) [inline opencog/moses/scoring/discriminating_bscore.cc]: The 3 new lines above can be removed.

File: opencog/moses/scoring/discriminating_bscore.cc [hunk 4]
```diff
@@ -510,7 +627,12 @@ behavioral_score prerec_bscore::operator()(const combo_tree& tr) const
             pos_total += ctr.positive_count;
         }
         // divide all element by pos_total (so it sums up to precision - 1/2)
-        boost::transform(bs, bs.begin(), arg1 / pos_total);
+
+        boost::transform(bs, bs.begin(),[pos_total](score_t x){
+                                        score_t zero = 0;
+                                        if(pos_total==0) return zero;
+                                        return x/pos_total;
```
Comment (ngeiswei) [inline opencog/moses/scoring/discriminating_bscore.cc]: You can simplify the body of the lambda by a 1-liner
```c++
return 0.0 < pos_total ? x / pos_total : 0.0;
```

File: opencog/moses/scoring/discriminating_bscore.cc [hunk 5]
```diff
@@ -560,6 +682,80 @@ behavioral_score prerec_bscore::operator()(const combo_tree& tr) const
...
+            // distict from non-positive ones)
+            bs.push_back((ctr.true_positive_sum - ctr.false_positive_sum)
+                         / 2);
+            pos_total += ctr.positive_count;
+        }
+        // divide all element by pos_total (so it sums up to precision - 1/2)
+        boost::transform(bs, bs.begin(),[pos_total](score_t x){
+                                        score_t zero = 0;
+                                        if(pos_total==0) return zero;
+                                        return x/pos_total;
```
Comment (ngeiswei) [inline opencog/moses/scoring/discriminating_bscore.cc]: Same 1-liner simplification as above.

File: tests/moses/scoring/discriminating_scoreUTest.cxxtest [hunk 6]
```diff
@@ -0,0 +1,155 @@
+/** discretize_contin_bscore_UTest.cxxtest ---
```
Comment (ngeiswei) [inline tests/moses/scoring/discriminating_scoreUTest.cxxtest]: Should match file name.

**Comment by ngeiswei** [comment]
@Eman22S I think you still need to resolve the conflicts and it should be OK.

**Pull request merged**